powerline/segments/common/bat.py

```diff
@@ -65,13 +70,20 @@ def _get_battery(pl):
 			cap_path = linux_bat_fmt.format(linux_bat)
 			if linux_bat.startswith('BAT') and os.path.exists(cap_path):
 				pl.debug('Using /sys/class/power_supply with battery {0}', linux_bat)
-
-				def _get_capacity(pl):
-					with open(cap_path, 'r') as f:
-						return int(float(f.readline().split()[0]))
-
-				return _get_capacity
-		pl.debug('Not using /sys/class/power_supply as no batteries were found')
+				with open(cap_path, 'r') as f:
+					_capacity = int(float(f.readline().split()[0]))
+
+		linux_ac_fmt = '/sys/class/power_supply/{0}/online'
+		for linux_ac in os.listdir('/sys/class/power_supply'):
```

Comment:
There is already an iteration over `/sys/class/power_supply`. You may use it to determine both BAT and AC directories.